src/openrct2/actions/PeepPickupAction.cpp

```diff
@@ -37,7 +37,7 @@ void PeepPickupAction::Serialise(DataSerialiser& stream)
 
 GameActions::Result PeepPickupAction::Query() const
 {
-    if (_spriteId >= MAX_ENTITIES || _spriteId == SPRITE_INDEX_NULL)
+    if (_spriteId.IsNull())
```

Comment:
In GuestNameAction, you did the exact opposite: `if (_spriteIndex.ToUnderlying() >= MAX_ENTITIES || _spriteIndex.IsNull())`

I think it’s best to determine which one is better and use that consistently.

Reply:
The underlying type is `uint16_t`, whose max is the same as `MAX_ENTITIES` (65535), so replacing `_spriteIndex.ToUnderlying() >= MAX_ENTITIES` with `_spriteIndex.IsNull()` is correct.

Reply:
I actually forgot to add the MAX_ENTITIES check there, there is no guarantee we will not change the type or limits at any point and this would avoid having to change code afterwards if we want arbitrary limits.